### ISSUE TITLE
Windows client failing mysteriously after upgrade

### ISSUE THREAD

**Issue by patricklucas** [keybase/client#5557]
I just upgraded from 0.1.15 to 0.1.16, per the suggestions on the website and CLI, and it appears to be quite broken.

When I attempt to log in (it auto-populated my username), beneath the 'Show Typing' checkbox on the login screen is the text 'Unknown RPC Error' in red with no other information.

If I click the tray icon and click 'Log In' I'm greeted with a screen with the header 'There was an error provisioning', but all the text is unselectable so I can't include it here without typing it all. (This is very very frustrating, please make this text selectable)

As I'm writing this issue report, I get a Windows notification that mentions something about a 'Read timeout' from the client.

There doesn't seem to be anything interesting in the app log:

```
From Keybase: log: Wed Jan 25 2017 18:53:03 GMT-0500 (Eastern Standard Time) (1485388383163): ["Dispatching action: window:changedFocus: ","{\"type\":\"window:changedFocus\",\"payload\":null}"]
From Keybase: log: Wed Jan 25 2017 18:53:03 GMT-0500 (Eastern Standard Time) (1485388383163): ["Diff: ",null]
From Keybase: log: Wed Jan 25 2017 18:53:12 GMT-0500 (Eastern Standard Time) (1485388392053): ["Dispatching action: window:changedFocus: ","{\"type\":\"window:changedFocus\",\"payload\":null}"]
From Keybase: log: Wed Jan 25 2017 18:53:12 GMT-0500 (Eastern Standard Time) (1485388392053): ["Diff: ",null]
From Keybase: log: Wed Jan 25 2017 18:53:14 GMT-0500 (Eastern Standard Time) (1485388394613): [{"code":914,"fields":null,"desc":""}]
From Keybase: log: Wed Jan 25 2017 18:53:14 GMT-0500 (Eastern Standard Time) (1485388394618): ["Dispatching action: login:waitingForResponse: ","{\"payload\":null,\"type\":\"login:waitingForResponse\"}"]
From Keybase: log: Wed Jan 25 2017 18:53:14 GMT-0500 (Eastern Standard Time) (1485388394618): ["Diff: ",null]
From Keybase: log: Wed Jan 25 2017 18:53:14 GMT-0500 (Eastern Standard Time) (1485388394623): ["Dispatching action: login:waitingForResponse: ","{\"payload\":null,\"type\":\"login:waitingForResponse\"}"]
From Keybase: log: Wed Jan 25 2017 18:53:14 GMT-0500 (Eastern Standard Time) (1485388394623): ["Diff: ",null]
From Keybase: log: Wed Jan 25 2017 18:53:14 GMT-0500 (Eastern Standard Time) (1485388394623): ["Dispatching action: login:loginDone: ","{\"type\":\"login:loginDone\",\"error\":true,\"payload\":null}"]
From Keybase: log: Wed Jan 25 2017 18:53:14 GMT-0500 (Eastern Standard Time) (1485388394623): ["Diff: ",null]
From Keybase: log: Wed Jan 25 2017 18:53:17 GMT-0500 (Eastern Standard Time) (1485388397262): ["Dispatching action: window:changedFocus: ","{\"type\":\"window:changedFocus\",\"payload\":null}"]
From Keybase: log: Wed Jan 25 2017 18:53:17 GMT-0500 (Eastern Standard Time) (1485388397262): ["Diff: ",null]
```

Any ideas?

**Comment by maxtaco** [comment]
Please issue `keybase log send` on the prompt

Cc: @zanderz

**Comment by patricklucas** [comment]
Done, log ID is `b292de974d86db1924a6ec1c`.

Interestingly, when I click the tray icon I now see the UI as if I were logged in (`private/`, `public/`, etc.) but `keybase track` in the CLI complains 'No secret key available' and I still see a login screen with 'Unknown RPC error' in the client window if I open it.

**Comment by zanderz** [comment]
We recently migrated the installation and settings path from %APPDATA%\Keybase to %LOCALAPPDATA%\Keybase. It is possible that we failed to move the keys correctly but the logs have rolled over past that. Do you see any files ending with mpack or json in %APPDATA%\Keybase ?

Another possibility is that your old client was not completely removed during the upgrade. Are there executable files in %APPDATA%\Keybase ?

**Comment by patricklucas** [comment]
There are indeed files remaining in `%APPDATA%\Keybase`, including `app-state.json`, `session.json`, and a `secretkeys.*.mpack` file, but no executables.

Would a manual fix be to move those files (or just the `.mpack`) to `%LOCALAPPDATA%\Keybase` and remove `%APPDATA%\Keybase`?

**Comment by zanderz** [comment]
It is a little more complicated, since the GUI still writes things in `%APPDATA%\Keybase`, such as `app-state.json`, and the service should be shut down before attempting it, but yes, that's the solution.
- From the Keybase Shell, issue `keybase ctl stop --kill-kbfs`
- Copy `session.json`, `config.json` and `*.mpack` from `%APPDATA%\Keybase` to `%LOCALAPPDATA%\Keybase`
- From the Keybase Shell, issue `start keybase ctl watchdog2` (or just reboot)
Sorry about the troubles, we won't change the install target again any time soon.

**Comment by patricklucas** [comment]
Hmm. That appeared to improve things at first, but after I typed my username, it wanted me to provision from another device, so it seems that it didn't pick up the secrets properly.

Then, I tried a `keybase id` from the shell, which errored with a timeout. I tried `keybase login` next, but that just hangs. Now back in the UI, if I type my username and click 'login' the button just spins.

After quitting Keybase from the tray icon, I was able to successfully `keybase id` from the shell, but `keybase encrypt <me>` just hangs. However, `keybase login` now asks for my username then again wants me to provision from another device.

Is there any way to just continue to use the device key from my previous installation, or do I need to revoke/reissue a device key for this device?
